docs/smart-contracts/modules.md

```diff
@@ -0,0 +1,564 @@
+          { "for": f, "against": (+ 1 a) })))
+  )
+```
+
+## Module properties and components
+
+As you've see module declarations start with the `module` keyword and a name.
+Module names must be unique within a namespace.
+You can define custom namespaces for local development.
+However, you must deploy modules to a registered namespace in the Kadena test or production networks.
+
+[Module declarations](/reference/syntax#module) use the following keywords to define the following module components:
+
+- [defun](/reference/syntax#defun) to define module functions.
+- [defschema](/reference/syntax#defschema) to define schemas for module tables.
+- [deftable](/reference/syntax#deftable) to define the tables to create for a module.
+- [defpact](/reference/syntax#defpact) to define multi-step transactions in the module.
+- [defconst](/reference/syntax#defconst) to define constant value variables in the module.
+- [defcap](/reference/syntax#defcap) to define capabilities in the module.
+- [use](/reference/syntax#use) to import functions from other modules into the module.
+- [implements](/reference/syntax#implements) to implement functions from interfaces exposed in other modules into the module.
+
+Modules can also include different types of [metadata](/reference/syntax/metadata), such as documentation strings or information for emitted events.
+
+When a module is declared, all references to native functions, interfaces, or definitions from other modules are resolved. 
+Resolution failure results in transaction rollback.
+
+### Module versioning and dependencies
+
+Module versioning is not supported, except by including a version identifier in the module name, for example, `accounts-v1` or `marmalade-v2`. 
+However, you can use _module hashes_ to import a specific version of a module with the [`use`](/reference/syntax#use) keyword.
+By specifying a module hash when you import the module, you can link your code to a particular release of a module that's identified by the hash.
+
+By combining module imports with module hashes, you can ensure that updated module code will fail to import if a dependent module has subsequently changed on the chain.
+
+### Module table creation
+
+Tables are created at the same time as modules and include the module name as a prefix to the table name.
+With this naming convention, the module acts as a guard to protect direct access to [database functions](/pact-5/database) that are controlled by the module owner.
+You can create any number of tables in a module.
+
+It’s important to note that you define table schemas, the tables that use the schemas, and the functions that insert, read, and modify database records inside of module code, but you create tables outside of module code.
+That is, tables are _defined_ in modules, but they are _created_ after the module declaration.
+This separation allows module code to be potentially updated without recreating the table. 
+
+## Interfaces
```

Comment:
Interface are not upgradeable

Reply:
Interfaces can't be upgraded and aren't governed by keysets or governance capability.